### DISCUSSION TITLE
Auto clear history

### DISCUSSION THREAD

**Discussion by Progression** [spatie/ray#744] [Ideas]
I leave Ray running all the time and after several days (weeks!) it can get pretty sluggish to scroll through or even for new items to show up. The "new Black Screen and Clear History" menu option clears things up nicely but it would be great if there was a way to have Ray run this automatically once a day/week or maybe after 100,200,1000 items have been added to the view. Alternatively a setting that prunes the view automatically so it only ever has a history of the last X items.

I think some kind of option like this would make Ray way more responsive.

**Comment by freekmurze** [comment]
I might indeed be nice to have a limit on how many history we keep. I feel like 20 screen should already be enough.

@Sam-Apostel could you look at this?

**Comment by Progression** [comment]
Personally I only use 1 screen so not having any issues with a limit on that, just the growth of the overall history. Capping that would be great 🙌
